.github/actions/install-aiida-workgraph/action.yaml

```
@@ -0,0 +1,45 @@
+name: Install aiida-workgraph
+description: Install aiida-workgraph package and its Python dependencies
+
+inputs:
+  python-version:
+    description: Python version
+    default: '3.9'              # Lowest supported version
+    required: false
+  extras:
+    description: list of optional dependencies
+    # NOTE: The default 'pre-commit' extra recursively contains
+    # other extras needed to run the tests.
+    default: pre-commit
+    required: false
+  # NOTE: Hard-learned lesson: we cannot use type=boolean here, apparently :-(
+  # https://stackoverflow.com/a/76294014
+  from-lock:
+    description: Install aiida-workgraph dependencies from uv lock file
+    default: 'true'
+    required: false
+  resolution-strategy:
+    description: uv dependency resolution strategy
+    default: highest
+    required: false
+
+runs:
+  using: composite
+  steps:
+  - name: Set Up Python
+    uses: actions/setup-python@v5
+    with:
+      python-version: ${{ inputs.python-version }}
+
+  - name: Set up uv
+    uses: astral-sh/setup-uv@v6
+    with:
+      version: 0.7.6
+      python-version: ${{ inputs.python-version }}
+      activate-environment: true
+
+  - name: Install dependencies from uv lock
+    if: ${{ inputs.from-lock == 'true' }}
+    # NOTE: We're asserting that the lockfile is up to date
+    run: uv sync --locked ${{ inputs.extras && format('--extra {0}', inputs.extras) || '' }}
```

It looks like you need to install both `pre-commit` and `tests` extras. It's easiest to just hard-code them.

```suggestion
    run: uv sync --locked --extra pre-commit --extra tests
```